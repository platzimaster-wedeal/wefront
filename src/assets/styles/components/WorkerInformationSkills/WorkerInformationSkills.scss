@import '../../partials/colors.scss';
@import '../../partials/sizes.scss';
@import '../../partials/fonts.scss';
@import '../../FontSize.scss';
@import '../../Breakpoints.scss';

.worker-information-skills {
  position: relative;
  max-width: 320px;
  width: auto;
  padding: $space-size-big;

  @include media(tablet){
    max-width: 50%;
  }
  
  &__title {
    margin: $space-size-extra-small 0;
    font-weight: 400;
    @include font-size($font-size-normal);

    @include media(tablet) {
      @include font-size($font-size-normal);
    }
  }

  &__list {
    display: flex;
    flex-flow: row wrap;
    
    li {
      margin: $space-size-small 0;
      margin-right: $space-size-small;
      @include font-size($font-size-small);

      @include media(tablet) {
        @include font-size($font-size-normal);
      }  
    }
  }
}