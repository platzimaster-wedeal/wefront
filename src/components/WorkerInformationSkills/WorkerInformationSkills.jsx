import React from "react";

// Styles
import "../../assets/styles/components/WorkerInformationSkills/WorkerInformationSkills.scss";

// Components
import Labels from "../Labels/Labels";

const WorkerInformationSkills = ({ skills }) => {
 return (
  <div className="worker-information-skills">
   <strong className="worker-information-skills__title">
    Skills And Categories
   </strong>
   <ul className="worker-information-skills__list">
    {skills && skills.length > 0 ? (
     skills.map((skill, id) => (
      <li key={id}>
       {" "}
       <Labels active name={skill} />{" "}
      </li>
     ))
    ) : (
     <span>This user aren't in some category.</span>
    )}
   </ul>
  </div>
 );
};

export default WorkerInformationSkills;
