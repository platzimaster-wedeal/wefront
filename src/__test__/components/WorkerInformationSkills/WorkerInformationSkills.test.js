import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import WorkerInformationSkills from "../../../components/WorkerInformationSkills/WorkerInformationSkills";

describe("Test for the component WorkerInformationSkills :D", () => {
 const WorkerInformationSkillsMounted = mount(
  <WorkerInformationSkills skills={["new skills"]} />
 );
 const WorkerInformationSkillsCreated = create(<WorkerInformationSkills />);

 describe("Visual tests for the component", () => {
  test("should the comopnent have the same tree", () => {
   expect(WorkerInformationSkillsCreated.toJSON()).toMatchSnapshot();
  });
 });

 describe("Functional tests for the component", () => {
  test("Should the component render one element", () => {
   expect(WorkerInformationSkillsMounted.length).toEqual(1);
  });

  test("Should the title of the component be a string", () => {
   expect(
    typeof WorkerInformationSkillsMounted.find(
     ".worker-information-skills__title"
    ).text()
   ).toEqual("string");
  });
  test("Should the title of the component be equal to Last Skills And Categories", () => {
   expect(
    WorkerInformationSkillsMounted.find(
     ".worker-information-skills__title"
    ).text()
   ).toEqual("Skills And Categories");
  });

  test("Should the description of null information be a string", () => {
   expect(
    typeof mount(<WorkerInformationSkills />)
     .find(".worker-information-skills__list span")
     .text()
   ).toEqual("string");
  });
  test("Should the description of the component be equal to This user aren't in some category.", () => {
   expect(
    mount(<WorkerInformationSkills />)
     .find(".worker-information-skills__list span")
     .text()
   ).toEqual("This user aren't in some category.");
  });
 });
});
