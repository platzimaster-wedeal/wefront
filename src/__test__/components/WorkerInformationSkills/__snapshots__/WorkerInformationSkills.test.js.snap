// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the component WorkerInformationSkills :D Visual tests for the component should the comopnent have the same tree 1`] = `
<div
  className="worker-information-skills"
>
  <strong
    className="worker-information-skills__title"
  >
    Skills And Categories
  </strong>
  <ul
    className="worker-information-skills__list"
  >
    <span>
      This user aren't in some category.
    </span>
  </ul>
</div>
`;
